### PR TITLE
Add ss02

### LINKED ISSUE #25113: Fix validation error type `SS02` and check in CI
#### Problem description
In order to have a continuous check by the CI on validation error `SS02` (Summary does not start with a capital letter), fixing them in the code base enables the addition to the CI for automated testing in the future.

`SS02` has just 61 occurrences (see [overview of the errors](https://gist.github.com/198cfa552c727d71bd7360fda66fdbb4)):

method | file | file_line | github_link
-- | -- | -- | --
pandas.tseries.offsets.DateOffset.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BusinessDay.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BusinessHour.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CustomBusinessDay.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CustomBusinessHour.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.MonthOffset.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.MonthEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.MonthBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BusinessMonthEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BusinessMonthBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CustomBusinessMonthEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CustomBusinessMonthBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.SemiMonthOffset.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.SemiMonthEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.SemiMonthBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Week.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.WeekOfMonth.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.LastWeekOfMonth.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.QuarterOffset.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BQuarterEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BQuarterBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.QuarterEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.QuarterBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.YearOffset.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BYearEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BYearBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.YearEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.YearBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.FY5253.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.FY5253Quarter.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
... | ... | ... | ...
pandas.tseries.offsets.Tick.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Day.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Hour.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Minute.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Second.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Milli.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Micro.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.Nano.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BDay.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BMonthEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.BMonthBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CBMonthEnd.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CBMonthBegin.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.tseries.offsets.CDay.normalize | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.ceil | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.combine | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.floor | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.fromordinal | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.fromtimestamp | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.replace | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.strftime | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timestamp.strptime | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.view | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.ceil | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.floor | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Timedelta.to_pytimedelta | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.HDFStore.groups | pandas/io/pytables.py | 1093.0 | https://github.com/pandas-dev/pandas/blob/master/pandas/io/pytables.py#L1093
pandas.Panel.items | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Panel.major_axis | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone
pandas.Panel.minor_axis | None | NaN | https://github.com/pandas-dev/pandas/blob/master/None#LNone

Todo:
* [ ]  get rid of the errors in the code base
* [ ]  update the `code_check.sh` script to take into account the `SS02` type of errors

### PR DESCRIPTION
**Pull request by abhishekmangla**
- [x] closes #25113, closes https://github.com/pandas-dev/pandas/issues/36745
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -207,6 +207,7 @@ level-by-level basis.
 
 Other enhancements
 ^^^^^^^^^^^^^^^^^^
+- Added SS02 check to ``code_checks.sh``. This will add a check in CI to make sure summary in docstrings begins with capital letter. (:issue:`25113`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:210]: this is not needed, can you remove

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -934,7 +934,7 @@ class Timestamp(_Timestamp):
         """
         Timestamp.combine(date, time)
 
-        date, time -> datetime with same date and time fields.
+        Date, time -> datetime with same date and time fields.
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx:937]: would revert this

File: pandas/_libs/tslibs/timestamps.pyx [hunk 3]
```diff
@@ -912,7 +912,7 @@ class Timestamp(_Timestamp):
         """
         Timestamp.fromtimestamp(ts)
 
-        timestamp[, tz] -> tz's local time from POSIX timestamp.
+        Timestamp[, tz] -> tz's local time from POSIX timestamp.
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx:915]: would revert this

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by abhishekmangla** [comment]
@jreback I resolved your comments, I think this is ready for review. Test that is failing in CI is pandas/tests/test_sorting.py which passes on my machine and is unrelated to this PR.

File: pandas/_libs/tslibs/nattype.pyx [hunk 4]
```diff
@@ -435,6 +434,21 @@ class NaTType(_NaT):
     # The remaining methods have docstrings copy/pasted from the analogous
     # Timestamp methods.
 
+    strftime = _make_error_func(
+        "strftime",
+        """
+        Timestamp.strftime(format)
+
+        Return a string representing the given POSIX timestamp
```
Comment (jreback) [inline pandas/_libs/tslibs/nattype.pyx:442]: can you add the link to the format strings (pointed at here: https://pandas.pydata.org/docs/reference/api/pandas.DatetimeIndex.strftime.html?highlight=strftime#pandas.DatetimeIndex.strftime)

Comment (abhishekmangla) [reply pandas/_libs/tslibs/nattype.pyx:442]: I have added the link

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @abhishekmangla

**Pull request merged**